runtime/lua/vim/treesitter.lua

```diff
@@ -288,15 +288,19 @@ function M.get_captures_at_pos(bufnr, row, col)
 
     local iter = q:query():iter_captures(root, buf_highlighter.bufnr, row, row + 1)
 
-    for id, node, metadata in iter do
+    for id, node, metadata, match in iter do
       if M.is_in_node_range(node, row, col) then
         ---@diagnostic disable-next-line: invisible
         local capture = q._query.captures[id] -- name of the capture in the query
         if capture ~= nil then
-          table.insert(
-            matches,
-            { capture = capture, metadata = metadata, lang = tree:lang(), id = id }
-          )
+          local _, pattern_i = match:info()
```

Comment:
```suggestion
          local _, pattern_id = match:info()
```

Reply:
(replying here for threading purposes)

> (Are the pattern ids stable?)

This is a good question and one I also had when I started implementing this... I haven't really found any specific documentation that indicates "stability" of these but from what I understand:

1. This pattern ID is really just an index, so given a query like `(foo) (bar) (baz)` you get IDs of `0`, `1`, `2`, respectively
	- Because nvim may concatenate queries together (e.g. `;extends`, `;inherits`), any changes in query file load order or similar mechanisms has the potential to change the indices
2. Pattern IDs are used in a number of public tree-sitter APIs (like `TSQueryMatch` and `ts_query_disable_pattern`), so I think they can probably be considered as stable as the rest of the API?